File: _tools/cmd/get.go
package cmd

import (
	"fmt"
	"os"
	"path/filepath"
	"sort"
	"strings"

	"github.com/atotto/clipboard"
	"github.com/spf13/cobra"

	"SecondBrain/_tools/util"
)

func init() {
	rootCmd.AddCommand(getCmd)
}

var getCmd = &cobra.Command{
	Use:     "get",
	Aliases: []string{"g"},
	Short:   "Get files from Desktop or Downloads",
	Args: func(cmd *cobra.Command, args []string) error {
		if len(args) > 3 {
			return fmt.Errorf("this command takes up to 3 arguments")
		}
		return nil
	},
	RunE: func(cmd *cobra.Command, args []string) error {
		if len(args) == 0 {
			listFiles()
		} else if len(args) == 1 {
			get(args[0], "")
		} else {
			get(args[0], args[1])
		}
		return nil
	},
}

func get(path, newName string) {
	id := util.GetID()

	basename := filepath.Base(path)
	extension := filepath.Ext(basename)
	name := strings.Replace(basename, extension, "", -1)
	if newName != "" {
		name = newName
	}

	filename := fmt.Sprintf("%s-%s%s", name, id, extension)
	newPath := fmt.Sprintf("_data/%s", filename)
	util.Exec(fmt.Sprintf("mv \"%s\" \"%s\"", path, newPath))
	markdownLink := fmt.Sprintf(fmt.Sprintf("![%s](../%s)", filename, newPath))
	err := clipboard.WriteAll(markdownLink)
	if err == nil {
		fmt.Printf("Copied to Clipboard: ")
	}
	fmt.Printf("%s\n", markdownLink)
}

func listFiles() {
	home := homeDir()
	desktopPath := fmt.Sprintf("%s/Desktop", home)
	downloadsPath := fmt.Sprintf("%s/Downloads", home)

	fmt.Printf("Desktop:\n")
	dirpath, _, filenames, _ := util.GetFilesAndDirectories(desktopPath)
	sort.Strings(filenames)
	for _, filename := range filenames {
		if filename[0:1] == "." {
			continue
		}
		fmt.Printf("\"%s/%s\"\n", dirpath, filename)
	}

	fmt.Println("")

	fmt.Printf("Downloads:\n")
	dirpath, _, filenames, _ = util.GetFilesAndDirectories(downloadsPath)
	sort.Strings(filenames)
	for _, filename := range filenames {
		if filename[0:1] == "." {
			continue
		}
		fmt.Printf("\"%s/%s\"\n", dirpath, filename)
	}
}

func homeDir() string {
	if h := os.Getenv("HOME"); h != "" {
		return h
	}
	return os.Getenv("USERPROFILE") // windows
}

File: _tools/Get.py
import sys
import datetime
import re
import os
import fileinput

from id import *

def PrintFiles():
    home = os.path.expanduser("~")
    desktopPath = "{}/Desktop".format(home)
    DownloadsPath = "{}/Downloads".format(home)

    print("Desktop:")
    dirpath, dirnames, filenames = ("", "", "")
    for (dirpath, dirnames, filenames) in os.walk(desktopPath):
        break
    filenames.sort()
    for filename in filenames:
        if filename[0] == ".":
            continue
        # filename = filename.replace(" ", "\ ")
        print("\"{}/{}\"".format(dirpath, filename))

    print("")

    print("Downloads:")
    dirpath, dirnames, filenames = ("", "", "")
    for (dirpath, dirnames, filenames) in os.walk(DownloadsPath):
        break
    filenames.sort()
    for filename in filenames:
        if filename[0] == ".":
            continue
        # filename = filename.replace(" ", "\ ")
        print("\"{}/{}\"".format(dirpath, filename))
    

def Get(filePath, newName=""):
    id = getID()

    basename = os.path.basename(filePath)
    name, extension = os.path.splitext(basename)
    if newName != "":
        name = newName
    filename = "{}-{}{}".format(name, id, extension)
    newPath = "../_data/{}".format(filename)
    os.system("mv \"{}\" \"{}\"".format(filePath, newPath))
    print("![{}]({})".format(filename, newPath))

if __name__ == "__main__":
    if len(sys.argv) > 3:
        print("Usage: ./get [filepath] [new_name]")
        sys.exit(1)
    if sys.argv[1] == "":
        PrintFiles()
        sys.exit(0)
    Get(sys.argv[1], sys.argv[2])
    sys.exit(0)
    

    
    